samcli/lib/samlib/resource_metadata_normalizer.py

```diff
@@ -270,3 +273,18 @@ def get_resource_id(resource_properties, logical_id):
             cdk_resource_id = cdk_resource_id[: -len(CDK_NESTED_STACK_RESOURCE_ID_SUFFIX)]
 
         return cdk_resource_id
+
+    @staticmethod
+    def normalize_build_properties(build_props) -> Dict:
```

Comment:
question - is Lambda Builder expecting snake case for build properties in general and not only esbuild?

Reply:
BuildProperties is a new field, but everything else we send lambda builders is snake case so I wanted to keep it consistent.